ophyd/signal.py

```diff
@@ -102,7 +102,6 @@ def __init__(self, *, name, value=0., timestamp=None, parent=None,
             timestamp = time.time()
 
         self._destroyed = False
-        self._finalizer = weakref.finalize(self, self.destroy)
```

Comment:
If I'm reading these changes correctly:

With this removed, there is no longer a mechanism by which `Signal.destroy` will be called automatically when a `Signal` (or any derived class) is torn down. It will now only happen if called manually.

Previous behavior was that `destroy` was effectively the destructor for `Signal` and derived classes, and called automatically... initially by `Signal.__del__` and then by this finalizer. If a derived class implemented its own `destroy`, it would get called automatically instead (and was responsible for propagating upward via `super`).

I just want to make sure I'm reading this right and that this is intended, since it seems like a pretty big departure from how Signal teardown used to work.

Reply:
Good point. Added release notes entry (which I should have done in any case) explaining the implementations of the change. Please review.